notus/scanner/scanner.py

```diff
@@ -131,16 +134,33 @@ def run_scan(
         """Handle the data necessary to start a scan,
         received via mqtt and run the scan."""
 
-        installed_packages = [
-            RPMPackage.from_full_name(name) for name in message.package_list
-        ]
+        advisory = self._loader.load_advisory(
+            operating_system=message.os_release
+        )
+        package_type_id = advisory.get("package_type", "") if advisory else ""
+        package_type = PackageType.from_string(package_type_id)
+        if not package_type:
+            logger.log(
+                logging.WARN, "%s invalid package type.", package_type_id
+            )
+            return
+        may_installed: Iterable[Optional[Package]] = (
+            DEBPackage.from_full_name(name)
+            if package_type == PackageType.DEB
+            else RPMPackage.from_full_name(name)
+            for name in message.package_list
+        )
+        installed_packages: Iterable[Package] = (
+            package for package in may_installed if package is not None
+        )
```

Comment:
What about packages filtered here? Is this an error/warning? Should the filtered packages be logged?

Reply:
It can only happen when the previous from_full_name fails which already does log a warning.

Reply:
added a comment for clarification